src/js/controller/controller.js

```diff
@@ -764,7 +764,11 @@ Object.assign(Controller.prototype, {
                     // Insert a small delay here so that other complete handlers can execute
                     resolved.then(_completeHandler);
                 }, _this)
-                .on(MEDIA_ERROR, _this.triggerError, _this);
+                .on(MEDIA_ERROR, _this.triggerError, _this)
+                .on(MEDIA_TYPE, (event) => {
+                    // Re-set mediaType so that it triggers change listeners even if queued
+                    _model.set('mediaType', event.mediaType);
```

Comment:
This adds "mediaType" back to the model which we moved to mediaModel intentionally.